pkg/virt-controller/services/renderresources.go

```diff
@@ -68,6 +69,21 @@ func WithEphemeralStorageRequest() ResourceRendererOption {
 	}
 }
 
+func WithoutDedicatedCPU(cpu *v1.CPU, cpuAllocationRatio int) ResourceRendererOption {
```

Comment:
I have tried to avoid that - the [`newResourceRenderer` template receiver function](https://github.com/kubevirt/kubevirt/pull/7975/files#diff-e2b17b934fcd4fb2278da4c106432d96e03f936e34e2dd4854843501aa719ad8R813) does receive the VMI, extracts relevant information from it, and sets the options accordingly.

And again... If I make the `newResourceRenderer` shorter and cleaner, I'll just move that logic into the constructor ... It's hard, because it can really go both ways, but I think the constructor should care about resources, not about VMIs.